### PR TITLE
Tighten up prehash section and fix spec bug

### PR DESCRIPTION
**Pull request by bwesterb**

### CODE REVIEW

**Review by jakemas — APPROVED** [review]

File: draft-ietf-lamps-dilithium-certificates.md [hunk 1]
```diff
@@ -394,18 +394,41 @@ in this section.
...
+# Pre-hashing (ExternalMu-ML-DSA) {#prehash}
+
+Some applications require prehashing, where the signature generation
+process can be separated into a pre-hash step and a core signature
+step in order to ease operational requirements around large or
+inconsistently-sized payloads. This can be performed at the
+protocol layer, but not all protocols support it.
+An example in [RFC5280] is the certificate revocation list
+(CRL) data structures, that do not include message digesting before signing.
+This makes signing large volumes of large CRLs challenging.
```
Comment (ounsworth) [inline draft-ietf-lamps-dilithium-certificates.md:406]: CRLs do tend to be large (up to 1 mb), but very rarely would you sign a lot of them at the same time. Typically a CA will only issue 1 CRL per day. That's why I also had the example of issuing a large number of certificates containing large public keys.

File: draft-ietf-lamps-dilithium-certificates.md [hunk 2]
```diff
@@ -394,18 +394,41 @@ in this section.
...
+
+As mentioned in the introduction, pure ML-DSA signing itself
+supports a prehashing flow by splitting the operation over two
+modules. In this section we make this "ExternalMu-ML-DSA"
+more explicit.
+
+There are two steps. First an `ExternalMu-ML-DSA.Prehash()`
+followed by `ExternalMu-ML-DSA.Sign()`. Together these operate
+as `ML-DSA.Sign()` from [FIPS204] and thus create exactly the same
+signatures as regular pure ML-DSA.
```
Comment (ounsworth) [inline draft-ietf-lamps-dilithium-certificates.md:416]: Suggest:

Together these are functionally equivalent to `ML-DSA.Sign()` from [FIPS204] in that they create exactly the same signatures as regular pure ML-DSA, which can be verified by the regular `ML-DSA.Verify()`.

File: draft-ietf-lamps-dilithium-certificates.md [hunk 3]
```diff
@@ -394,18 +394,41 @@ in this section.
...
+Note that ExternalMu-ML-DSA describes a different signature API from ML-DSA
+and therefore might require explicit support from hardware or
+software cryptographic modules.
+
+Note that the signing mode defined here is different from HashML-DSA
+defined in [FIPS204] section 5.4. This specification uses exclusively
+ExternalMu-ML-DSA for pre-hashed use cases, and thus HashML-DSA as
+defined in [FIPS204] and identified by `id-hash-ml-dsa-44-with-sha512`,
+`id-hash-ml-dsa-65-with-sha512`, and `id-hash-ml-dsa-87-with-sha512`
+MUST NOT be used in X.509 and related PKIX protocols.
```
Comment (ounsworth) [inline draft-ietf-lamps-dilithium-certificates.md:428]: Suggest:

... public keys identified by these OIDs MUST NOT be used within X.509 certificates, and the signature algorithms identified by these OIDs MUST NOT be used within X.509 and related PKIX protocols.

**Review by ounsworth — APPROVED** [review]
PR is an improvement. Could be merged as-is, but I left some suggestions.

File: draft-ietf-lamps-dilithium-certificates.md (hunk 1 above)
Comment (bwesterb) [reply draft-ietf-lamps-dilithium-certificates.md:406]: Will tweak in separate PR.

File: draft-ietf-lamps-dilithium-certificates.md (hunk 2 above)
Comment (bwesterb) [reply draft-ietf-lamps-dilithium-certificates.md:416]: Will tweak in separate PR.

File: draft-ietf-lamps-dilithium-certificates.md (hunk 3 above)
Comment (bwesterb) [reply draft-ietf-lamps-dilithium-certificates.md:428]: Will tweak in separate PR.

File: draft-ietf-lamps-dilithium-certificates.md (hunk 1 above)
Comment (bwesterb) [reply draft-ietf-lamps-dilithium-certificates.md:406]: https://github.com/lamps-wg/dilithium-certificates/pull/55

**Pull request merged**